### PR TITLE
Merge api modules

### PR DESCRIPTION
**Pull request by robobario**
### Type of change

- Enhancement / new feature

### Description

We found these are always used together and think it's less work to release them as a single module.

### Checklist

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious-api/pom.xml [hunk 1]
```diff
@@ -18,24 +18,18 @@
...
     <packaging>jar</packaging>
 
     <properties>
         <libs.dir>libs</libs.dir>
     </properties>
 
-    <name>Kroxylicious Filter API</name>
-    <description>The public interfaces, classes, etc required to author custom filters and configure Kroxylicious to use them</description>
+    <name>Kroxylicious API</name>
+    <description>generic APIs used throughout Kroxylicious</description>
```
Comment (SamBarker) [inline kroxylicious-api/pom.xml:30]: Nit: not sure generic is a great term here.

```suggestion
    <description>The public interfaces, classes, etc required to build filters and interact programatically with Kroxylicious</description>
```

File: scripts/release-api.sh [hunk 2]
```diff
@@ -22,7 +22,6 @@ fi
 
 echo "Setting API version to ${RELEASE_API_VERSION}"
 mvn -q versions:set -DnewVersion="${RELEASE_API_VERSION}" -DprocessAllModules=true -pl ":kroxylicious-api" -DgenerateBackupPoms=false
```
Comment (SamBarker) [inline scripts/release-api.sh:24]:
```suggestion
mvn -q versions:set -DnewVersion="${RELEASE_API_VERSION}" -DprocessAllModules=true -pl ${API_MODULES} -DgenerateBackupPoms=false
```

Though arguably we should just do away with the variable.

**Review by SamBarker — APPROVED** [review]
LGTM

File: kroxylicious-api/pom.xml (hunk 1 above)
Comment (robobario) [reply kroxylicious-api/pom.xml:30]: Yeah good idea to update it, this was the description from kroxylicious-api

**Pull request merged**